Compile error in build

When I build after install mavsdk libmavsdk-dev_3.7.1_ubuntu24.04_amd64.deb file this code there are many compile errors as below.
Do you know root cause?

project/siyi-a8-mini-camera-manager/camera-manager/camera_manager.cpp: In function ‘int main(int, char**)’:
project/siyi-a8-mini-camera-manager/camera-manager/camera_manager.cpp:121:73: error: ‘ComponentType’ is not a member of ‘mavsdk::Mavsdk’
  121 |     mavsdk::Mavsdk mavsdk{mavsdk::Mavsdk::Configuration{mavsdk::Mavsdk::ComponentType::Camera}};
      |                                                                         ^~~~~~~~~~~~~
project/siyi-a8-mini-camera-manager/camera-manager/camera_manager.cpp:121:73: error: no matching function for call to ‘mavsdk::Mavsdk::Configuration::Configuration(<brace-enclosed initializer list>)’
In file included from project/siyi-a8-mini-camera-manager/camera-manager/camera_manager.cpp:7:
/usr/include/mavsdk/mavsdk.h:206:18: note: candidate: ‘mavsdk::Mavsdk::Configuration::Configuration(mavsdk::ComponentType)’
  206 |         explicit Configuration(ComponentType component_type);
      |                  ^~~~~~~~~~~~~
/usr/include/mavsdk/mavsdk.h:206:18: note:   conversion of argument 1 would be ill-formed:
/usr/include/mavsdk/mavsdk.h:199:18: note: candidate: ‘mavsdk::Mavsdk::Configuration::Configuration(uint8_t, uint8_t, bool)’
  199 |         explicit Configuration(
      |                  ^~~~~~~~~~~~~
/usr/include/mavsdk/mavsdk.h:199:18: note:   candidate expects 3 arguments, 1 provided
/usr/include/mavsdk/mavsdk.h:190:11: note: candidate: ‘constexpr mavsdk::Mavsdk::Configuration::Configuration(const mavsdk::Mavsdk::Configuration&)’
  190 |     class Configuration {
      |           ^~~~~~~~~~~~~
/usr/include/mavsdk/mavsdk.h:190:11: note:   conversion of argument 1 would be ill-formed:
project/siyi-a8-mini-camera-manager/camera-manager/camera_manager.cpp:121:56: error: expected primary-expression before ‘{’ token
  121 |     mavsdk::Mavsdk mavsdk{mavsdk::Mavsdk::Configuration{mavsdk::Mavsdk::ComponentType::Camera}};
      |                                                        ^
project/siyi-a8-mini-camera-manager/camera-manager/camera_manager.cpp:121:56: error: expected ‘}’ before ‘{’ token
  121 |     mavsdk::Mavsdk mavsdk{mavsdk::Mavsdk::Configuration{mavsdk::Mavsdk::ComponentType::Camera}};
      |                          ~                             ^
project/siyi-a8-mini-camera-manager/camera-manager/camera_manager.cpp:121:56: error: no matching function for call to ‘mavsdk::Mavsdk::Mavsdk(<brace-enclosed initializer list>)’
/usr/include/mavsdk/mavsdk.h:290:5: note: candidate: ‘mavsdk::Mavsdk::Mavsdk(Configuration)’
  290 |     Mavsdk(Configuration configuration);
      |     ^~~~~~
/usr/include/mavsdk/mavsdk.h:290:5: note:   conversion of argument 1 would be ill-formed:
project/siyi-a8-mini-camera-manager/camera-manager/camera_manager.cpp:121:56: error: expected ‘,’ or ‘;’ before ‘{’ token
  121 |     mavsdk::Mavsdk mavsdk{mavsdk::Mavsdk::Configuration{mavsdk::Mavsdk::ComponentType::Camera}};
      |                                                        ^
project/siyi-a8-mini-camera-manager/camera-manager/camera_manager.cpp: At global scope:
project/siyi-a8-mini-camera-manager/camera-manager/camera_manager.cpp:126:27: error: expected constructor, destructor, or type conversion before ‘(’ token
  126 |     mavsdk::log::subscribe([](mavsdk::log::Level level,   // message severity level
      |                           ^
project/siyi-a8-mini-camera-manager/camera-manager/camera_manager.cpp:151:6: error: expected unqualified-id before ‘)’ token
  151 |     });
      |      ^
project/siyi-a8-mini-camera-manager/camera-manager/camera_manager.cpp:153:5: error: expected unqualified-id before ‘for’
  153 |     for (auto& connection : parser.connections) {
      |     ^~~
project/siyi-a8-mini-camera-manager/camera-manager/camera_manager.cpp:167:57: error: ‘ComponentType’ is not a member of ‘mavsdk::Mavsdk’
  167 |         mavsdk.server_component_by_type(mavsdk::Mavsdk::ComponentType::Camera)};
      |                                                         ^~~~~~~~~~~~~
project/siyi-a8-mini-camera-manager/camera-manager/camera_manager.cpp:167:79: error: no matching function for call to ‘mavsdk::FtpServer::FtpServer(<brace-enclosed initializer list>)’
  167 |         mavsdk.server_component_by_type(mavsdk::Mavsdk::ComponentType::Camera)};
      |                                                                               ^
In file included from project/siyi-a8-mini-camera-manager/camera-manager/camera_manager.cpp:10:
/usr/include/mavsdk/plugins/ftp_server/ftp_server.h:89:5: note: candidate: ‘mavsdk::FtpServer::FtpServer(const mavsdk::FtpServer&)’
   89 |     FtpServer(const FtpServer& other);
      |     ^~~~~~~~~
/usr/include/mavsdk/plugins/ftp_server/ftp_server.h:89:5: note:   conversion of argument 1 would be ill-formed:
/usr/include/mavsdk/plugins/ftp_server/ftp_server.h:41:14: note: candidate: ‘mavsdk::FtpServer::FtpServer(std::shared_ptr<mavsdk::ServerComponent>)’
   41 |     explicit FtpServer(std::shared_ptr<ServerComponent> server_component);
      |              ^~~~~~~~~
/usr/include/mavsdk/plugins/ftp_server/ftp_server.h:41:14: note:   conversion of argument 1 would be ill-formed:
project/siyi-a8-mini-camera-manager/camera-manager/camera_manager.cpp:166:40: error: expected primary-expression before ‘{’ token
  166 |     auto ftp_server = mavsdk::FtpServer{
      |                                        ^
project/siyi-a8-mini-camera-manager/camera-manager/camera_manager.cpp:171:5: error: expected unqualified-id before ‘if’
  171 |     if (!std::filesystem::exists(path)) {
      |     ^~
project/siyi-a8-mini-camera-manager/camera-manager/camera_manager.cpp:175:10: error: ‘cout’ in namespace ‘std’ does not name a type
  175 |     std::cout << "Using FTP root: " << path << " to serve camera xml file." << std::endl;
      |          ^~~~
In file included from project/siyi-a8-mini-camera-manager/camera-manager/camera_manager.cpp:1:
/usr/include/c++/13/iostream:63:18: note: ‘std::cout’ declared here
   63 |   extern ostream cout;          ///< Linked to standard output
      |                  ^~~~
project/siyi-a8-mini-camera-manager/camera-manager/camera_manager.cpp:178:5: error: expected unqualified-id before ‘if’
  178 |     if (ftp_result != mavsdk::FtpServer::Result::Success) {
      |     ^~
project/siyi-a8-mini-camera-manager/camera-manager/camera_manager.cpp:184:57: error: ‘ComponentType’ is not a member of ‘mavsdk::Mavsdk’
  184 |         mavsdk.server_component_by_type(mavsdk::Mavsdk::ComponentType::Camera)};
      |                                                         ^~~~~~~~~~~~~
project/siyi-a8-mini-camera-manager/camera-manager/camera_manager.cpp:184:79: error: no matching function for call to ‘mavsdk::ParamServer::ParamServer(<brace-enclosed initializer list>)’
  184 |         mavsdk.server_component_by_type(mavsdk::Mavsdk::ComponentType::Camera)};
      |                                                                               ^
In file included from project/siyi-a8-mini-camera-manager/camera-manager/camera_manager.cpp:11:
/usr/include/mavsdk/plugins/param_server/param_server.h:328:5: note: candidate: ‘mavsdk::ParamServer::ParamServer(const mavsdk::ParamServer&)’
  328 |     ParamServer(const ParamServer& other);
      |     ^~~~~~~~~~~
/usr/include/mavsdk/plugins/param_server/param_server.h:328:5: note:   conversion of argument 1 would be ill-formed:
/usr/include/mavsdk/plugins/param_server/param_server.h:41:14: note: candidate: ‘mavsdk::ParamServer::ParamServer(std::shared_ptr<mavsdk::ServerComponent>)’
   41 |     explicit ParamServer(std::shared_ptr<ServerComponent> server_component);
      |              ^~~~~~~~~~~
/usr/include/mavsdk/plugins/param_server/param_server.h:41:14: note:   conversion of argument 1 would be ill-formed:
project/siyi-a8-mini-camera-manager/camera-manager/camera_manager.cpp:183:44: error: expected primary-expression before ‘{’ token
  183 |     auto param_server = mavsdk::ParamServer{
      |                                            ^
project/siyi-a8-mini-camera-manager/camera-manager/camera_manager.cpp:187:5: error: expected unqualified-id before ‘switch’
  187 |     switch (siyi_camera.resolution()) {
      |     ^~~~~~
project/siyi-a8-mini-camera-manager/camera-manager/camera_manager.cpp:200:5: error: expected unqualified-id before ‘switch’
  200 |     switch (siyi_camera.codec(siyi::Camera::Type::Stream)) {
      |     ^~~~~~
project/siyi-a8-mini-camera-manager/camera-manager/camera_manager.cpp:209:5: error: ‘param_server’ does not name a type
  209 |     param_server.provide_param_int("CAM_MODE", 0);
      |     ^~~~~~~~~~~~
project/siyi-a8-mini-camera-manager/camera-manager/camera_manager.cpp:210:5: error: ‘param_server’ does not name a type
  210 |     param_server.provide_param_int("STREAM_RES", stream_res);
      |     ^~~~~~~~~~~~
project/siyi-a8-mini-camera-manager/camera-manager/camera_manager.cpp:211:5: error: ‘param_server’ does not name a type
  211 |     param_server.provide_param_int("STREAM_BITRATE", static_cast<int32_t>(siyi_camera.bitrate()));
      |     ^~~~~~~~~~~~
project/siyi-a8-mini-camera-manager/camera-manager/camera_manager.cpp:212:5: error: ‘param_server’ does not name a type
  212 |     param_server.provide_param_int("STREAM_CODEC", stream_codec);
      |     ^~~~~~~~~~~~
project/siyi-a8-mini-camera-manager/camera-manager/camera_manager.cpp:214:5: error: ‘param_server’ does not name a type
  214 |     param_server.subscribe_changed_param_int([&](auto param_int) {
      |     ^~~~~~~~~~~~
project/siyi-a8-mini-camera-manager/camera-manager/camera_manager.cpp:242:6: error: expected unqualified-id before ‘)’ token
  242 |     });
      |      ^
project/siyi-a8-mini-camera-manager/camera-manager/camera_manager.cpp:245:57: error: ‘ComponentType’ is not a member of ‘mavsdk::Mavsdk’
  245 |         mavsdk.server_component_by_type(mavsdk::Mavsdk::ComponentType::Camera)};
      |                                                         ^~~~~~~~~~~~~
project/siyi-a8-mini-camera-manager/camera-manager/camera_manager.cpp:245:79: error: no matching function for call to ‘mavsdk::CameraServer::CameraServer(<brace-enclosed initializer list>)’
  245 |         mavsdk.server_component_by_type(mavsdk::Mavsdk::ComponentType::Camera)};
      |                                                                               ^
In file included from project/siyi-a8-mini-camera-manager/camera-manager/camera_manager.cpp:9:
/usr/include/mavsdk/plugins/camera_server/camera_server.h:1050:5: note: candidate: ‘mavsdk::CameraServer::CameraServer(const mavsdk::CameraServer&)’
 1050 |     CameraServer(const CameraServer& other);
      |     ^~~~~~~~~~~~
/usr/include/mavsdk/plugins/camera_server/camera_server.h:1050:5: note:   conversion of argument 1 would be ill-formed:
/usr/include/mavsdk/plugins/camera_server/camera_server.h:41:14: note: candidate: ‘mavsdk::CameraServer::CameraServer(std::shared_ptr<mavsdk::ServerComponent>)’
   41 |     explicit CameraServer(std::shared_ptr<ServerComponent> server_component);
      |              ^~~~~~~~~~~~
/usr/include/mavsdk/plugins/camera_server/camera_server.h:41:14: note:   conversion of argument 1 would be ill-formed:
project/siyi-a8-mini-camera-manager/camera-manager/camera_manager.cpp:244:46: error: expected primary-expression before ‘{’ token
  244 |     auto camera_server = mavsdk::CameraServer{
      |                                              ^
project/siyi-a8-mini-camera-manager/camera-manager/camera_manager.cpp:261:5: error: expected unqualified-id before ‘if’
  261 |     if (ret != mavsdk::CameraServer::Result::Success) {
      |     ^~
project/siyi-a8-mini-camera-manager/camera-manager/camera_manager.cpp:266:5: error: ‘ret’ does not name a type
  266 |     ret = camera_server.set_video_streaming(mavsdk::CameraServer::VideoStreaming{
      |     ^~~
project/siyi-a8-mini-camera-manager/camera-manager/camera_manager.cpp:269:6: error: expected unqualified-id before ‘)’ token
  269 |     });
      |      ^
project/siyi-a8-mini-camera-manager/camera-manager/camera_manager.cpp:271:5: error: expected unqualified-id before ‘if’
  271 |     if (ret != mavsdk::CameraServer::Result::Success) {
      |     ^~
project/siyi-a8-mini-camera-manager/camera-manager/camera_manager.cpp:278:5: error: ‘camera_server’ does not name a type
  278 |     camera_server.subscribe_take_photo([&](int32_t index) {
      |     ^~~~~~~~~~~~~
project/siyi-a8-mini-camera-manager/camera-manager/camera_manager.cpp:308:10: error: expected unqualified-id before ‘)’ token
  308 |         });
      |          ^
project/siyi-a8-mini-camera-manager/camera-manager/camera_manager.cpp:313:5: error: ‘camera_server’ does not name a type
  313 |     camera_server.subscribe_start_video([&](int32_t) {
      |     ^~~~~~~~~~~~~
project/siyi-a8-mini-camera-manager/camera-manager/camera_manager.cpp:330:6: error: expected unqualified-id before ‘)’ token
  330 |     });
      |      ^
project/siyi-a8-mini-camera-manager/camera-manager/camera_manager.cpp:332:5: error: ‘camera_server’ does not name a type
  332 |     camera_server.subscribe_stop_video([&](int32_t) {
      |     ^~~~~~~~~~~~~
project/siyi-a8-mini-camera-manager/camera-manager/camera_manager.cpp:346:6: error: expected unqualified-id before ‘)’ token
  346 |     });
      |      ^
project/siyi-a8-mini-camera-manager/camera-manager/camera_manager.cpp:348:5: error: ‘camera_server’ does not name a type
  348 |     camera_server.subscribe_set_mode(
      |     ^~~~~~~~~~~~~
project/siyi-a8-mini-camera-manager/camera-manager/camera_manager.cpp:363:6: error: expected unqualified-id before ‘)’ token
  363 |     });
      |      ^
project/siyi-a8-mini-camera-manager/camera-manager/camera_manager.cpp:366:5: error: ‘camera_server’ does not name a type
  366 |     camera_server.subscribe_capture_status([&](int32_t) {
      |     ^~~~~~~~~~~~~
project/siyi-a8-mini-camera-manager/camera-manager/camera_manager.cpp:383:6: error: expected unqualified-id before ‘)’ token
  383 |     });
      |      ^
project/siyi-a8-mini-camera-manager/camera-manager/camera_manager.cpp:385:5: error: ‘camera_server’ does not name a type
  385 |     camera_server.subscribe_storage_information([&](int32_t) {
      |     ^~~~~~~~~~~~~
project/siyi-a8-mini-camera-manager/camera-manager/camera_manager.cpp:399:6: error: expected unqualified-id before ‘)’ token
  399 |     });
      |      ^
project/siyi-a8-mini-camera-manager/camera-manager/camera_manager.cpp:401:5: error: ‘camera_server’ does not name a type
  401 |     camera_server.subscribe_zoom_range([&](float zoom_factor) {
      |     ^~~~~~~~~~~~~
project/siyi-a8-mini-camera-manager/camera-manager/camera_manager.cpp:415:6: error: expected unqualified-id before ‘)’ token
  415 |     });
      |      ^
project/siyi-a8-mini-camera-manager/camera-manager/camera_manager.cpp:417:5: error: ‘camera_server’ does not name a type
  417 |     camera_server.subscribe_zoom_in_start([&](int) {
      |     ^~~~~~~~~~~~~
project/siyi-a8-mini-camera-manager/camera-manager/camera_manager.cpp:420:6: error: expected unqualified-id before ‘)’ token
  420 |     });
      |      ^
project/siyi-a8-mini-camera-manager/camera-manager/camera_manager.cpp:422:5: error: ‘camera_server’ does not name a type
  422 |     camera_server.subscribe_zoom_out_start([&](int) {
      |     ^~~~~~~~~~~~~
project/siyi-a8-mini-camera-manager/camera-manager/camera_manager.cpp:425:6: error: expected unqualified-id before ‘)’ token
  425 |     });
      |      ^
project/siyi-a8-mini-camera-manager/camera-manager/camera_manager.cpp:427:5: error: ‘camera_server’ does not name a type
  427 |     camera_server.subscribe_zoom_stop([&](int) {
      |     ^~~~~~~~~~~~~
project/siyi-a8-mini-camera-manager/camera-manager/camera_manager.cpp:430:6: error: expected unqualified-id before ‘)’ token
  430 |     });
      |      ^
project/siyi-a8-mini-camera-manager/camera-manager/camera_manager.cpp:433:5: error: expected unqualified-id before ‘while’
  433 |     while (true) {
      |     ^~~~~
project/siyi-a8-mini-camera-manager/camera-manager/camera_manager.cpp:437:5: error: expected unqualified-id before ‘return’
  437 |     return 0;
      |     ^~~~~~
project/siyi-a8-mini-camera-manager/camera-manager/camera_manager.cpp:438:1: error: expected declaration before ‘}’ token
  438 | }
      | ^
gmake[2]: *** [CMakeFiles/camera_manager.dir/build.make:76: CMakeFiles/camera_manager.dir/camera_manager.cpp.o] 오류 1
gmake[1]: *** [CMakeFiles/Makefile2:143: CMakeFiles/camera_manager.dir/all] 오류 2
gmake: *** [Makefile:146: all] 오류 2

--- a/.github/workflows/build.yml
+++ b/.github/workflows/build.yml
@@ -1,4 +1,4 @@
-name: Build for RPi
+name: Build
 
 on:
   push:
@@ -11,6 +11,35 @@ on:
     - '*'
 
 jobs:
+  build-ubuntu-x64:
+    name: Build on Ubuntu 24.04 x64
+    runs-on: ubuntu-24.04
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          submodules: recursive
+      - name: Install dependencies
+        run: |
+          sudo apt-get update
+          sudo apt-get install -y libgstreamer1.0-dev libgstrtspserver-1.0-dev build-essential cmake git wget
+          wget https://github.com/mavlink/MAVSDK/releases/download/v3.7.1/libmavsdk-dev_3.7.1_debian12_amd64.deb
+          sudo dpkg -i libmavsdk-dev_3.7.1_debian12_amd64.deb
+          rm libmavsdk-dev_3.7.1_debian12_amd64.deb
+      - name: Build project
+        run: |
+          cmake -DCMAKE_BUILD_TYPE=RelWithDebInfo -DBUILD_SHARED_LIBS=ON -DCMAKE_INSTALL_PREFIX=install -Bbuild -S.
+          cmake --build build -j$(nproc) --target install
+      - name: Upload build artifacts
+        uses: actions/upload-artifact@v4
+        with:
+          name: ubuntu-x64-build
+          path: |
+            install/
+      - name: Run tests
+        run: |
+          cd camera-manager
+          ctest --test-dir ../build/camera-manager
+
   create-deb-package:
     name: Create .deb package
     runs-on: ubuntu-24.04

--- a/.github/workflows/rpi-build.yml
+++ b/.github/workflows/rpi-build.yml
@@ -33,9 +33,9 @@ jobs:
             apt-get update
             apt-get install -y libgstreamer1.0-dev libgstrtspserver-1.0-dev build-essential cmake git wget rubygems
             gem install fpm
-            wget https://github.com/mavlink/MAVSDK/releases/download/v2.10.2/libmavsdk-dev_2.10.2_debian12_arm64.deb
-            dpkg -i libmavsdk-dev_2.10.2_debian12_arm64.deb
-            rm libmavsdk-dev_2.10.2_debian12_arm64.deb
+            wget https://github.com/mavlink/MAVSDK/releases/download/v3.7.1/libmavsdk-dev_3.7.1_debian12_arm64.deb
+            dpkg -i libmavsdk-dev_3.7.1_debian12_arm64.deb
+            rm libmavsdk-dev_3.7.1_debian12_arm64.deb
             git config --global --add safe.directory "$GITHUB_WORKSPACE"
             cmake -DCMAKE_BUILD_TYPE=RelWithDebInfo -DBUILD_SHARED_LIBS=ON -DCMAKE_INSTALL_PREFIX=install -Bbuild -S.
             cmake --build build -j$(nproc) --target install

--- a/.github/workflows/rpi-build.yml
+++ b/.github/workflows/rpi-build.yml
@@ -41,7 +41,7 @@ jobs:
             cmake --build build -j$(nproc) --target install
             ./create_packages.sh ./install . arm64 siyi-a8-mini-camera-manager
       - name: Upload package
-        uses: actions/upload-artifact@v3
+        uses: actions/upload-artifact@v4
         with:
           name: deb-package
           path: |

--- a/.github/workflows/rpi-build.yml
+++ b/.github/workflows/rpi-build.yml
@@ -13,7 +13,7 @@ on:
 jobs:
   create-deb-package:
     name: Create .deb package
-    runs-on: ubuntu-22.04
+    runs-on: ubuntu-24.04
     permissions:
       contents: write
       packages: write
@@ -21,13 +21,12 @@ jobs:
       - uses: actions/checkout@v4
         with:
           submodules: recursive
-      - uses: uraimo/run-on-arch-action@v2
+      - uses: uraimo/run-on-arch-action@v3
         name: Run commands
         id: runcmd
         with:
-          arch: none
-          distro: none
-          base_image: arm64v8/debian:bookworm
+          arch: aarch64
+          distro: bookworm
           githubToken: ${{ github.token }}
           run: |
             apt-get update

--- a/README.md
+++ b/README.md
@@ -168,11 +168,11 @@ The camera manager is implemented using A small application on top of MAVSDK the
 
 ### Get MAVSDK
 
-Download the latest MAVSDK release as a .deb. For Raspberry Pi 4 running a 64bit image, this is using the package for the `arm64` architecture:
+Download the latest MAVSDK v3 release as a .deb. For Raspberry Pi 4 running a 64bit image, this is using the package for the `arm64` architecture:
 
 ```
-wget https://github.com/mavlink/MAVSDK/releases/download/v2.12.12/libmavsdk-dev_2.12.12_debian12_arm64.deb
-sudo dpkg -i libmavsdk-dev_2.12.12_debian12_arm64.deb
+wget https://github.com/mavlink/MAVSDK/releases/download/v3.7.1/libmavsdk-dev_3.7.1_debian12_arm64.deb
+sudo dpkg -i libmavsdk-dev_3.7.1_debian12_arm64.deb
 ```
 
 ### Build

--- a/README.md
+++ b/README.md
@@ -283,3 +283,5 @@ This could have a couple of reasons:
 
 - Are both processes running, the rtsp-rebroadcast as well as the camera-manager?
 - Are all arguments to camera-manager correct?
+- Is everything H265 or H264? The rtsp_rebroadcast has H265 hard-coded, so the setting would have to match it.
+  Note that the camera's RTSP path is always `main.264` even if it is set to H265.

--- a/README.md
+++ b/README.md
@@ -1,5 +1,7 @@
 # SIYI A8 mini camera manager
 
+[![Build](https://github.com/julianoes/siyi-a8-mini-camera-manager/actions/workflows/build.yml/badge.svg?branch=main)](https://github.com/julianoes/siyi-a8-mini-camera-manager/actions/workflows/build.yml)
+
 Camera Manager for SIYI A8 mini for RPi based on MAVSDK.
 
 ## Prerequisites

--- a/camera-manager/camera_manager.cpp
+++ b/camera-manager/camera_manager.cpp
@@ -118,7 +118,7 @@ int main(int argc, char* argv[])
     }
 
     // MAVSDK setup second
-    mavsdk::Mavsdk mavsdk{mavsdk::Mavsdk::Configuration{mavsdk::Mavsdk::ComponentType::Camera}};
+    mavsdk::Mavsdk mavsdk{mavsdk::Mavsdk::Configuration{mavsdk::ComponentType::Camera}};
 
     // We overwrite the mavsdk logs to prepend "Mavsdk:" and to make sure we flush it after every
     // line using std::endl. Otherwise, it gets buffered by systemctl and logs appear delayed.
@@ -164,7 +164,7 @@ int main(int argc, char* argv[])
     }
 
     auto ftp_server = mavsdk::FtpServer{
-        mavsdk.server_component_by_type(mavsdk::Mavsdk::ComponentType::Camera)};
+        mavsdk.server_component_by_type(mavsdk::ComponentType::Camera)};
 
     // If running locally when built first, otherwise use system-wise:
     std::string path = "./camera-manager/mavlink_ftp_root";
@@ -181,7 +181,7 @@ int main(int argc, char* argv[])
     }
 
     auto param_server = mavsdk::ParamServer{
-        mavsdk.server_component_by_type(mavsdk::Mavsdk::ComponentType::Camera)};
+        mavsdk.server_component_by_type(mavsdk::ComponentType::Camera)};
 
     int32_t stream_res = 0;
     switch (siyi_camera.resolution()) {
@@ -242,7 +242,7 @@ int main(int argc, char* argv[])
     });
 
     auto camera_server = mavsdk::CameraServer{
-        mavsdk.server_component_by_type(mavsdk::Mavsdk::ComponentType::Camera)};
+        mavsdk.server_component_by_type(mavsdk::ComponentType::Camera)};
 
     auto ret = camera_server.set_information({
         .vendor_name = "SIYI",

